flagsmith/flagsmith.py

```diff
@@ -11,18 +11,22 @@
 
 
 class Flagsmith:
-    def __init__(self, environment_id, api=SERVER_URL):
+    def __init__(self, environment_id, api=SERVER_URL, custom_headers=None):
         """
         Initialise Flagsmith environment.
 
         :param environment_id: environment key obtained from the Flagsmith UI
         :param api: (optional) api url to override when using self hosted version
+        :param custom_headers: (optional) dict which will be passed in headers for each api call
         """
         self.environment_id = environment_id
         self.api = api
         self.flags_endpoint = api + FLAGS_ENDPOINT
         self.identities_endpoint = api + IDENTITY_ENDPOINT
         self.traits_endpoint = api + TRAIT_ENDPOINT
+        self.custom_headers = (
+            custom_headers if custom_headers and type(custom_headers) == dict else {}
```

Comment:
I don't think we should _really_ cater for people passing in custom_headers that aren't in dict format. That would simplify this line to `self.custom_headers = custom_headers or {}`.

Reply:
✅